From 9b9d4cbe67f4271d01339660c285c90570621d22 Mon Sep 17 00:00:00 2001
From: Pedro Palazon Candel <pedro@joyent.com>
Date: Thu, 30 Aug 2018 18:17:52 +0200
Subject: [PATCH] TRITON-591 refreservation not set properly on bhyve disks
 Reviewed by: Mike Gerdts <mike.gerdts@joyent.com> Reviewed by: Josh Wilsdon
 <josh@wilsdon.ca> Approved by: Josh Wilsdon <josh@wilsdon.ca>

---
 src/vm/node_modules/VM.js | 21 +++------------------
 1 file changed, 3 insertions(+), 18 deletions(-)

diff --git a/src/vm/node_modules/VM.js b/src/vm/node_modules/VM.js
index 5c6d76f3..f7b97338 100644
--- a/src/vm/node_modules/VM.js
+++ b/src/vm/node_modules/VM.js
@@ -2106,7 +2106,6 @@ function createVolume(volume, opts, callback)
     assert.string(opts.brand, 'opts.brand');
     assert.object(opts.log, 'opts.log');
 
-    var brand = opts.brand;
     var log = opts.log;
     var refreserv;
     var size;
@@ -2133,23 +2132,7 @@ function createVolume(volume, opts, callback)
 
     if (volume.hasOwnProperty('refreservation')) {
         refreserv = volume.refreservation;
-    } else if (volume.hasOwnProperty('image_uuid')) {
-        // We can't set refreserv > volsize when making a clone, so set it
-        // to volsize instead.
-        refreserv = size;
-        log.debug({
-            size: size
-        }, 'refreservation unset, setting refreserv=size for cloned disk');
-    } else if (brand === 'kvm') {
-        // For KVM we keep the pre-OS-6753 for now to not break compatibility
-        // but could change this to match bhyve (default case below) in the
-        // future.
-        refreserv = size;
-        log.debug({
-            size: size
-        }, 'refreservation unset for kvm, setting refreserv=size for new disk');
     } else {
-        // We default to letting SmartOS set refreservation for bhyve: OS-6753
         log.debug('refreservation unset, letting system determine');
         refreserv = 'auto';
     }
@@ -2203,7 +2186,9 @@ function createVolume(volume, opts, callback)
                     args.push('-o', 'volblocksize='
                         + volume.block_size);
                 }
-                if (refreserv !== 'auto') {
+                if (refreserv === 'auto') {
+                    args.push('-o', 'refreservation=auto');
+                } else {
                     args.push('-o', 'refreservation=' + refreserv + 'M');
                 }
                 args.push(snapshot, target);
-- 
2.21.0

